src/interpreter/vm.cc

```diff
@@ -67,30 +67,28 @@ namespace verona::interpreter
   void VM::execute_finaliser(VMObject* object)
   {
     const VMDescriptor* descriptor = object->descriptor();
-    uint32_t finaliser_ip = descriptor->finaliser_ip;
-    if (finaliser_ip != 0)
-    {
-      auto vm = VM::local_vm;
-      vm->trace("Finaliser for: {}", descriptor->name);
+    assert(descriptor->finaliser_ip > 0);
```

Comment:
Nice change to make this code non-conditional by making the VM do the conditional work.